activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -77,7 +79,7 @@ def parse_raw_value_as_a_number(raw_value)
       end
 
       def is_integer?(raw_value)
-        /\A[+-]?\d+\z/ === raw_value.to_s
```

Comment:
@southwolf can you compare with - `\A[+-]?\d+(\.)?(?(1)0+\z|\z)`, and which one is faster.